presto-bytecode/src/main/java/com/facebook/presto/bytecode/DumpBytecodeVisitor.java

```diff
@@ -0,0 +1,615 @@
+
+        return null;
+    }
+
+    @Override
+    public Void visitIf(BytecodeNode parent, IfStatement ifStatement)
+    {
+        if (ifStatement.getComment() != null) {
+            printLine();
+            printLine("// %s", ifStatement.getComment());
+        }
+        printLine("if {");
+        indentLevel++;
+        visitNestedNode("condition", ifStatement.condition(), ifStatement);
+        if (!ifStatement.ifTrue().isEmpty()) {
+            visitNestedNode("ifTrue", ifStatement.ifTrue(), ifStatement);
+        }
+        if (!ifStatement.ifFalse().isEmpty()) {
+            visitNestedNode("ifFalse", ifStatement.ifFalse(), ifStatement);
+        }
+        indentLevel--;
+        printLine("}");
+        return null;
+    }
+
+    @Override
+    public Void visitFor(BytecodeNode parent, ForLoop forLoop)
+    {
+        if (forLoop.getComment() != null) {
+            printLine();
+            printLine("// %s", forLoop.getComment());
+        }
+        printLine("for {");
+        indentLevel++;
+        visitNestedNode("initialize", forLoop.initialize(), forLoop);
+        visitNestedNode("condition", forLoop.condition(), forLoop);
+        visitNestedNode("update", forLoop.update(), forLoop);
+        visitNestedNode("body", forLoop.body(), forLoop);
+        indentLevel--;
+        printLine("}");
+        return null;
+    }
+
+    @Override
+    public Void visitWhile(BytecodeNode parent, WhileLoop whileLoop)
+    {
+        if (whileLoop.getComment() != null) {
+            printLine();
+            printLine("// %s", whileLoop.getComment());
+        }
+        printLine("while {");
+        indentLevel++;
+        visitNestedNode("condition", whileLoop.condition(), whileLoop);
+        visitNestedNode("body", whileLoop.body(), whileLoop);
+        indentLevel--;
+        printLine("}");
+        return null;
+    }
+
+    @Override
+    public Void visitDoWhile(BytecodeNode parent, DoWhileLoop doWhileLoop)
+    {
+        if (doWhileLoop.getComment() != null) {
+            printLine();
+            printLine("// %s", doWhileLoop.getComment());
+        }
+        printLine("while {");
+        indentLevel++;
+        visitNestedNode("body", doWhileLoop.body(), doWhileLoop);
+        visitNestedNode("condition", doWhileLoop.condition(), doWhileLoop);
+        indentLevel--;
+        printLine("}");
+        return null;
+    }
+
+    @Override
+    public Void visitSwitch(BytecodeNode parent, SwitchStatement switchStatement)
+    {
+        if (switchStatement.getComment() != null) {
+            printLine();
+            printLine("// %s", switchStatement.getComment());
+        }
+        printLine("switch {");
+        indentLevel++;
+        visitNestedNode("expression", switchStatement.expression(), switchStatement);
+        for (CaseStatement caseStatement : switchStatement.cases()) {
+            visitNestedNode(format("case %s:", caseStatement.getKey()), caseStatement.getBody(), switchStatement);
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This code uses &#39;%s&#39; to format int: &#39;getKey&#39; expression. This is a potential locale-sensitive handling issue. It might cause errors in the handling and processing of the statement at line: 405. Consider formatting this data with &#39;%d&#39; instead.